src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java

```diff
@@ -88,6 +91,26 @@ public ResponseEntity<LiveEventDTO> getLiveEvents(@PathVariable String liveEvent
         return ResponseEntity.ok(data);
     }
 
+    @GetMapping("/vod")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_USER', 'ROLE_LEVEL_1', 'ROLE_LEVEL_2', 'ROLE_LEVEL_3', 'ROLE_LEVEL_4')")
+    public ResponseEntity<PlaybackDTO> getVod(@RequestParam UUID recordingId,
+                                              @RequestParam String userId,
+                                              @RequestParam(required = false) String mediaService) {
+        // todo: dont rely on naming convention, link asset name in db
+        IMediaService resolvedMediaService;
```

Comment:
It may be cleaner to move this override logic into the MediaServiceBroker. You could override the method there which accepts a string which can return the service for that string or fallback to the actual one configured?